rust/src/user/user_manager.rs

```diff
@@ -81,119 +49,107 @@ impl UserManager {
+    /// driver.users.create(username, password).await;
     /// ```
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
-    pub async fn get(&self, username: impl Into<String>) -> Result<Option<User>> {
-        let username = username.into();
-        self.run_any_node(|server_connection: ServerConnection| {
-            let username = username.clone();
-            async move { server_connection.get_user(username).await }
-        })
-        .await
+    pub async fn create(&self, username: impl Into<String>, password: impl Into<String>) -> Result {
+        let uname = username.into();
+        let passwd = password.into();
+        let mut error_buffer = Vec::with_capacity(self.server_connections.len());
+        for (server_id, server_connection) in self.server_connections.iter() {
+            match server_connection.create_user(uname.clone(), passwd.clone()).await {
+                Ok(res) => return Ok(res),
+                Err(err) => error_buffer.push(format!("- {}: {}", server_id, err))
+            }
+        }
+        Err(ConnectionError::ServerConnectionFailedWithError { error: error_buffer.join("\n") })?
     }
 
-    /// Sets a new password for a user. This operation can only be performed by administrators.
+    /// Deletes a user with the given name.
     ///
     /// # Arguments
     ///
-    /// * `username` -- The name of the user to set the password of
-    /// * `password` -- The new password
+    /// * `username` -- The name of the user to be deleted
     ///
     /// # Examples
     ///
     /// ```rust
-    /// driver.users.password_set(username, password).await;
+    /// driver.users.delete(username).await;
     /// ```
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
-    pub async fn set_password(&self, username: impl Into<String>, password: impl Into<String>) -> Result {
-        let username = username.into();
-        let password = password.into();
-        self.run_any_node(|server_connection: ServerConnection| {
-            let username = username.clone();
-            let password = password.clone();
-            async move { server_connection.set_user_password(username, password).await }
-        })
-        .await
-    }
-
-    #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
-    async fn run_any_node<F, P, R>(&self, task: F) -> Result<R>
-    where
-        F: Fn(ServerConnection) -> P,
-        P: Future<Output = Result<R>>,
-    {
-        // if !self.connection.is_cloud() {
-        //     Err(Error::Connection(ConnectionError::UserManagementCloudOnly))
-        // } else {
-        //     DatabaseManager::new(self.connection.clone())
-        //         .get(Self::SYSTEM_DB)
-        //         .await?
-        //         .run_failsafe(|database| task(database.connection().clone()))
-        //         .await
-        // }
-        todo!()
+    pub async fn delete(&self, username: impl Into<String>) -> Result {
+        let uname = username.into();
```

Comment:
Not a fan of `uname`, we're ok to use `let username = username.into()` in Rust code

Reply:
Fixed in 52387df94315ee79060d6d189f22e28077db045b